### PR TITLE
[CI:DOCS] Improve swagger+CORS metadata docs

### LINKED ISSUE #6466: APIv2 documentation is not working
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
https://podman.readthedocs.io/en/latest/_static/api.html does not work. I saw this week or two weeks ago for the first time so not sure if still down or again.

wentwrong

### PR DESCRIPTION
**Pull request by cevich**
Fixes #6466 - The CORS metadata only refererred to `http://docs.podman.io` and not `https://docs.podman.io`.  CORS treats these as completely different sites.

Note: For client (user's web-browser) security, maintaining the CORS configuration should not be automated.  Doing so risks exposing admin access to the storage bucket, permitting malicious CORS metadata to be set and content served.

Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

File: contrib/cirrus/README.md [hunk 1]
```diff
@@ -167,33 +167,65 @@ env:
...
+
+However, when the CORS metadata is missing or incorrectly configured, clients will receive an
+error-message similar to:
+
+![Javascript Stack Trace Image](swagger_stack_trace.png)
+
+For documentation built by Read The Docs from the master branch, CORS metadata is
+set on the `libpod-master-releases` storage bucket.  Viewing or setting the CORS
+metadata on the bucket requires having locally [installed and
+configured the goole-cloud SDK](https://cloud.google.com/sdk/docs).  It also requires having
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/README.md:194]: goole-cloude or google-cloud?

File: contrib/cirrus/README.md [hunk 2]
```diff
@@ -167,33 +167,65 @@ env:
...
     }
 ]
 ```
 
+```
+gsutil cors set /path/to/file.json gs://libpod-master-releases
+```
+
+***Note:*** The CORS metadata does _NOT_ change after the `docs` task uploads a new swagger YAML
+file.  Therefor, if it is not functioning or misconfigured, a person must have altered it or
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/README.md:226]:
```suggestion
file.  Therefore, if it is not functioning or misconfigured, a person must have altered it or
```

File: contrib/cirrus/README.md [hunk 3]
```diff
@@ -167,33 +167,65 @@ env:
...
+changes were made to the refering site (e.g. `docs.podman.io`).
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/README.md:227]:
```suggestion
changes were made to the referring site (e.g. `docs.podman.io`).
```

File: contrib/cirrus/README.md [hunk 4]
```diff
@@ -167,33 +167,65 @@ env:
...
-PR number or branch name.
-
-***Note***: [The online documentation](http://docs.podman.io/en/latest/_static/api.html)
-is presented through javascript on the client-side.  This requires CORS to be properly
-configured on the bucket, for the `http://docs.podman.io` origin.  Please see
-[Configuring CORS on a bucket](https://cloud.google.com/storage/docs/configuring-cors#configure-cors-bucket)
-for details.  This may be performed by anybody with admin access to the google storage bucket,
-using the following JSON:
+bucket.](https://storage.googleapis.com/libpod-pr-releases/) for testing purposes
+only.  For the master branche, a [separate bucket is
```
Comment (edsantiago) [inline contrib/cirrus/README.md:175]: s/branche/branch/

File: contrib/cirrus/README.md [hunk 5]
```diff
@@ -167,33 +167,65 @@ env:
...
+used](https://storage.googleapis.com/libpod-master-releases) and provides the
+content rendered on [the API Reference page](https://docs.podman.io/en/latest/_static/api.html)
+
+The online API reference, is presented by javascript to the client.  To prevent hijacking
```
Comment (edsantiago) [inline contrib/cirrus/README.md:179]: s/,//

File: docs/Readme.md [hunk 6]
```diff
@@ -30,10 +30,26 @@ link on that page.
...
+Since there are multiple systems and local cache is involved, it's possible that updates to
+documentation (especially the swagger/API docs) will lag by 10-or-so minutes.  However,
+because the client (i.e. your web browser) is fetching content from multiple locations that
+do not share a common domain, accessing the API section may show a stack-trace similar to
+the following:
+
+![Javascript Stack Trace Image](../contrib/cirrus/swagger_stack_trace.png)
+
+If reloading the page, or clearing your local cache does not fix the problem, it is
+it is likely caused by broken metadata needed to protect clients from cross-site-scripting
```
Comment (edsantiago) [inline docs/Readme.md:51]: s/it is //

**Comment by edsantiago** [comment]
LGTM once corrections are made. Thank you, the 'SDK' thing was helpful in understanding how to address this. But thanks most of all for finding and fixing the root cause of the problem.

File: contrib/cirrus/README.md (hunk 1 above)
Comment (cevich) [reply contrib/cirrus/README.md:194]: goole-cloud is the new version :laughing:

**Comment by cevich** [comment]
Thanks @edsantiago @TomSweeneyRedHat for the review, I'll make those corrections shortly.

@baude or anyone else recall a http -> https change on the RTD side while I was on "Vacation"?   That's definitely what broke, (we didn't allow https previously).  Knowing that it was not caused by a sun-spot or other high-energy particle would help me feel better in sticking with a manual CORS-metadata maintenance scheme.

Otherwise, I can automate this, despite my security-hairs standing on end.  I'm just not sure it will do much overall, beyond requiring even more documentation and adding more complexity :confused:

**Comment by TomSweeneyRedHat** [comment]
@cevich The http -> https target is painted on my back.  That just became a possibility in ReadTheDocs, they didn't support it until a week or two ago and I flipped the switch.  So no aliens creeping around GitHub for that one.

File: contrib/cirrus/README.md (hunk 1 above)
Comment (cevich) [reply contrib/cirrus/README.md:194]: ood rief!   My 'g' key cap is literally held on by a blob of silicon lue,  as all replacements and repairs are on hold by IT :sa: 

Okay I smashed the key a bit harder, now this is fixed.

File: contrib/cirrus/README.md (hunk 3 above)
Comment (cevich) [reply contrib/cirrus/README.md:227]: Hmm, is it correct both ways?  My spell-checker missed this :confused:

File: docs/Readme.md (hunk 6 above)
Comment (cevich) [reply docs/Readme.md:51]: good catch.

File: contrib/cirrus/README.md (hunk 3 above)
Comment (edsantiago) [reply contrib/cirrus/README.md]: It's so commonly misspelled that some spell checkers may have given up on it. The incorrect form is most famously embedded in Internet lore as the [HTTP_REFERER](https://tools.ietf.org/html/rfc2616#section-14.36) header.

**Comment by cevich** [comment]
Thanks for the honesty @TomSweeneyRedHat, no worries it's a completely understandable change.  Likely my bad for not including https in the list to begin with.  I'm just glad it didn't break because of some automation or file change (or aliens) :grinning:

**Comment by rhatdan** [comment]
@cevich Whenever you want to cancel your hold...

**Comment by cevich** [comment]
thx guys

**Pull request merged**